crates/parser/src/helpers.rs

```diff
@@ -58,3 +59,20 @@ impl<A, D> ReadableInstruction<A, D> {
 pub trait InstructionParser<C> {
     fn parse_ix(_: &InstructionUpdate) -> Result<C, String>;
 }
+
+pub fn from_coption_to_option<T>(coption: COption<T>) -> Option<T> {
+    match coption {
+        COption::Some(val) => Some(val),
+        COption::None => None,
+    }
+}
+
+pub trait IntoProtoData<O> {
+    fn into_proto_data(self) -> O;
+}
+
+pub fn pubkey_to_string(pubkey: SolanaPubkey) -> String { pubkey.to_string() }
+
+pub fn bytes_to_elgamal_pubkey(bytes: [u8; ELGAMAL_KEYPAIR_LEN / 2]) -> String {
+    bs58::encode(bytes).into_string()
+}
```

Comment:
Since L63, 74, 76 are type conversion represent them using the From trait.

Reply:
these types are from external crate so i guess we cant impl From Trait

Reply:
SolanaPubkey(pubkey::Pubkey)
ElgamalPubkeyBytes([u8; ELGAMAL_KEYPAIR_LEN / 2])

and then define from traits on these structs.

Reply:
fixed